### PR TITLE
Implement GetSpec in the config API

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

What does this PR do? Why is it needed?

This PR implements config API's `GetSpec` method according to the spec: https://ethereum.github.io/eth2.0-APIs/#/Config/getSpec

Which issues(s) does this PR fix?

Part of #7510

Other notes for review

N/A

### CODE REVIEW

**Review by rauljordan — APPROVED** [review]
ouch that looks like it was so painful to write!

File: beacon-chain/rpc/beaconv1/config.go [hunk 1]
```diff
@@ -17,11 +21,80 @@ func (bs *Server) GetForkSchedule(ctx context.Context, req *ptypes.Empty) (*ethp
...
+	return &ethpb.SpecResponse{Data: data}, nil
 }
 
 // GetDepositContract retrieves deposit contract address and genesis fork version.
 func (bs *Server) GetDepositContract(ctx context.Context, req *ptypes.Empty) (*ethpb.DepositContractResponse, error) {
 	return nil, errors.New("unimplemented")
 }
+
+func prepareConfigSpec() map[string]string {
+	data := make(map[string]string)
```
Comment (terencechain) [inline beacon-chain/rpc/beaconv1/config.go:38]: How can we ensure that 
1.) when a person adds new config and this gets updated accordingly
2.) the spec name is correctly aligned

This is going to get fragmented. Can we utilize yaml tags as keys?

**Review by terencechain — CHANGES_REQUESTED** [review]

File: beacon-chain/rpc/beaconv1/config.go (hunk 1 above)
Comment (rauljordan) [reply beacon-chain/rpc/beaconv1/config.go:38]: Good point. We can use reflect to walk through the struct fields of the config for a future-proof solution

File: beacon-chain/rpc/beaconv1/config.go [hunk 2]
```diff
@@ -35,3 +49,37 @@ func (bs *Server) GetDepositContract(ctx context.Context, req *ptypes.Empty) (*e
...
+		case reflect.Slice:
+			data[tag] = hexutil.Encode(vField.Bytes())
+		case reflect.Array:
+			data[tag] = hexutil.Encode(reflect.ValueOf(&config).Elem().Field(i).Slice(0, vField.Len()).Bytes())
+		case reflect.String:
+			data[tag] = vField.String()
+		case reflect.Uint8:
+			data[tag] = hexutil.Encode([]byte{uint8(vField.Uint())})
+		default:
+			return nil, fmt.Errorf("unsupported config field type: %s", vField.Kind().String())
```
Comment (rkapka) [inline beacon-chain/rpc/beaconv1/config.go:80]: Cannot be tested as I am not able to change the type of a field during the test.

File: beacon-chain/rpc/beaconv1/config.go [hunk 3]
```diff
@@ -19,8 +25,15 @@ func (bs *Server) GetForkSchedule(ctx context.Context, req *ptypes.Empty) (*ethp
...
 // - all other values are returned as number.
-func (bs *Server) GetSpec(ctx context.Context, req *ptypes.Empty) (*ethpb.SpecResponse, error) {
-	return nil, errors.New("unimplemented")
+func (bs *Server) GetSpec(ctx context.Context, _ *ptypes.Empty) (*ethpb.SpecResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "beaconV1.GetSpec")
+	defer span.End()
+
+	data, err := prepareConfigSpec()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to prepare spec data: %v", err)
```
Comment (rkapka) [inline beacon-chain/rpc/beaconv1/config.go:34]: Cannot be tested as I am not able to make `prepareConfigSpec` fail.

File: shared/params/config.go [hunk 4]
```diff
@@ -16,83 +16,83 @@ type BeaconChainConfig struct {
...
-	MinGenesisTime                 uint64 `yaml:"MIN_GENESIS_TIME"`                   // MinGenesisTime is the time that needed to pass before kicking off beacon chain.
-	TargetAggregatorsPerCommittee  uint64 `yaml:"TARGET_AGGREGATORS_PER_COMMITTEE"`   // TargetAggregatorsPerCommittee defines the number of aggregators inside one committee.
-	HysteresisQuotient             uint64 `yaml:"HYSTERESIS_QUOTIENT"`                // HysteresisQuotient defines the hysteresis quotient for effective balance calculations.
-	HysteresisDownwardMultiplier   uint64 `yaml:"HYSTERESIS_DOWNWARD_MULTIPLIER"`     // HysteresisDownwardMultiplier defines the hysteresis downward multiplier for effective balance calculations.
-	HysteresisUpwardMultiplier     uint64 `yaml:"HYSTERESIS_UPWARD_MULTIPLIER"`       // HysteresisUpwardMultiplier defines the hysteresis upward multiplier for effective balance calculations.
+	TargetCommitteeSize            uint64 `yaml:"TARGET_COMMITTEE_SIZE" api:"target_committee_size"`                           // TargetCommitteeSize is the number of validators in a committee when the chain is healthy.
```
Comment (terencechain) [inline shared/params/config.go:19]: Can't we just use the current `yaml` tag and convert it to lower case to be used as key?

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
Looks good, awaiting @terencechain 's approval

**Review by terencechain — APPROVED** [review]

**Pull request merged**